### PR TITLE
Rip out deterministic torch that could be slowing things down

### PR DESCRIPTION
**Pull request by smorad**
## Why are these changes needed?

In RLlib, torch has known to be slower than tensorflow. [One reason for this](https://pytorch.org/docs/stable/notes/randomness.html#reproducibility) could be the fact that we force determinism when using torch, yet not when using tensorflow. Since rllib is still not deterministic (try running cartpole-ppo twice and see if you get the same results), we might as well treat torch like tf and remove this code.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by avnishn — COMMENTED** [review]
I'm overall fine with this change since we can't really guarantee seed reproducibility right now anyways

**Comment by gjoliver** [comment]
if there are <= 1 remote workers, things should be deterministic right?
why do you think we don't enforce this for tf? we literally try to do that right below?

**Comment by smorad** [comment]
@gjoliver we do not. We set the seed, we do NOT enable determinism for TF. CUDA will select different kernels based on the inputs when deterministic mode is false. GPU operations occur out-of-order and are non-deterministic in most cases, every for very simple operations like sum: https://github.com/pytorch/pytorch/issues/75240

**Comment by gjoliver** [comment]
> @gjoliver we do not. We set the seed, we do NOT enable determinism for TF. CUDA will select different kernels based on the inputs when deterministic mode is false. GPU operations occur out-of-order and are non-deterministic in most cases, every for very simple operations like sum: [pytorch/pytorch#75240](https://github.com/pytorch/pytorch/issues/75240)
> 
> Setting deterministic mode means the GPU is very rigid in how it is able to execute commands, and which kernels it can use.

actually @kouroshHakha recently made at least ppo deterministic on GPU.
the trick is to set min_train_timesteps_per_iteration instead of min_time_s_per_iteration if you need determinism.
do you still want to merge this PR if that's the case?
